drivers/storage/portworx/portworx.go

```diff
@@ -652,6 +654,42 @@ func defaultPortworxImageVersion(releases *manifest.ReleaseManifest) string {
 	return defaultPortworxVersion
 }
 
+func setNodeSpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
+	if len(toUpdate.Spec.Nodes) == 0 {
+		return
+	}
+
+	updatedNodeSpecs := make([]corev1alpha1.NodeSpec, 0)
+	for _, nodeSpec := range toUpdate.Spec.Nodes {
+		nodeSpecCopy := nodeSpec.DeepCopy()
+		if nodeSpec.Storage == nil {
+			nodeSpecCopy.Storage = toUpdate.Spec.Storage.DeepCopy()
+		} else if toUpdate.Spec.Storage != nil {
+			if nodeSpecCopy.Storage.UseAll == nil && toUpdate.Spec.Storage.UseAll != nil {
```

Comment:
UseAll | UseAllWithPartitions | DeviceSpecs are exclusive and only one of them should be set.